plugins/magic.py

```diff
@@ -1,39 +1,67 @@
-"""!magic <search term> return the first magic card with name containing <search term>"""
+"""!magic <search term> return magic card specified by the search term"""
 
+import math
 import re
 import random
 from urllib import quote
 
 import requests
 
-def magic(searchterm):
-    quotedsearchterm = quote(searchterm)
-    url = "https://api.deckbrew.com/mtg/cards?name={}"
-    url = url.format(quotedsearchterm)
-    results = requests.get(url).json()
+filterlist = ["type","supertype","subtype","name","oracle","set","rarity","set","color","multicolor","m","format","status"]
 
-    if not results:
+def search(searchterm):
```

Comment:
Looks like we quote() searchterm a few times. Can we just do it once at the beginning?